packages/batch-submitter/src/exec/run-batch-submitter.ts

```diff
@@ -276,4 +278,36 @@ export const run = async () => {
   if (requiredEnvVars.RUN_STATE_BATCH_SUBMITTER === 'true') {
     loop(() => stateBatchSubmitter.submitNextBatch())
   }
+
```

Comment:
What do you think of creating a class like so:

```js
class MetricsServer {
  constructor(options) {
    // assign particular options to this
    // validate options
  }
  async start() {}
  async stop() {}
}
```

Then that can be imported here and `start` can be called

Also keeping it lean is nice but without tests its hard to know if this will work. I wouldn't be opposed to using express since we use it in the DTL

Reply:
i think express would work! i'd like to leave this server easy to take apart/rebuild until our dashboards are up, since metrics as a whole is still taking form and this just serves that purpose

Reply:
> this just serves that purpose

With no tests there is no proof that the purpose is served?

I'd argue that having a simple class would make the server 10x easier to take apart and rebuild because to be honest I'd have to hack around a bit to figure out exactly what is going on with the low level `http` package in nodejs

Reply:
Yeah, I feel like this is potentially reusable code, IMO would be good to break this out into a small class and add a few basic tests.

Reply:
Here is an example of a more functional approach: https://github.com/graphprotocol/common-ts/blob/5544327c0388a73f4e627625d077b64352031f9f/packages/common-ts/src/metrics/index.ts#L30

Reply:
thoughts on using the graph's implementation in this iteration?^ seems to serve our purpose

Reply:
We can use the graph's implementation for now, what do you think about putting it in `core-utils` and then it can be imported into each typescript service?

Reply:
Ideally we have a configurable http host/port as well as an flag to enable it optionally

Reply:
just updated this to use express and put it in common-ts